networkx/generators/tests/test_degree_seq.py

```diff
@@ -197,18 +203,44 @@ def test_directed_havel_hakimi():
     pytest.raises(nx.exception.NetworkXError, nx.directed_havel_hakimi_graph, din, dout)
 
 
-def test_degree_sequence_tree():
-    z = [1, 1, 1, 1, 1, 2, 2, 2, 3, 4]
-    G = nx.degree_sequence_tree(z)
-    assert len(G) == len(z)
-    assert len(list(G.edges())) == sum(z) / 2
-
-    pytest.raises(
-        nx.NetworkXError, nx.degree_sequence_tree, z, create_using=nx.DiGraph()
-    )
-
-    z = [1, 1, 1, 1, 1, 1, 2, 2, 2, 3, 4]
-    pytest.raises(nx.NetworkXError, nx.degree_sequence_tree, z)
+@pytest.mark.parametrize(
+    "deg_seq",
+    [
+        [0],
+        [1, 1],
+        [2, 2, 2, 1, 1],
+        [3, 1, 1, 1],
+        [4, 1, 1, 1, 1],
+        [1, 1, 1, 1, 1, 2, 2, 2, 3, 4],
+    ],
+)
+def test_degree_sequence_tree(deg_seq):
+    G = nx.degree_sequence_tree(deg_seq)
+    assert len(G) == len(deg_seq)
+    assert G.number_of_edges() == sum(deg_seq) // 2
```

Comment:
Maybe test that the degree sequence is correct instead of len and sum?
Seomthing like:  `assert sorted(dict(G.degree).values()) == sorted(deg_seq)`

Reply:
Agreed, that would be better. I'll do this in a follow-up PR while cleaning up some other tests in this file. Thanks for the suggestion!

Reply:
Done in #8257 :)